python/llm/src/bigdl/llm/transformers/model.py

```diff
@@ -121,6 +120,19 @@ def from_pretrained(cls,
         optimize_model = kwargs.pop("optimize_model", True)
         user_quantization_config = kwargs.pop("quantization_config", None)
 
+        if user_quantization_config is not None and \
+                "BitsAndBytesConfig" in str(user_quantization_config.__class__):
+            if user_quantization_config.bnb_4bit_quant_type is not None:
+                load_in_low_bit = user_quantization_config.bnb_4bit_quant_type
+            else:
+                load_in_low_bit = "sym_int4"
+            if user_quantization_config.bnb_4bit_compute_dtype is not None:
+                kwargs["torch_dtype"] = user_quantization_config.bnb_4bit_compute_dtype
+            else:
+                kwargs["torch_dtype"] = torch.float32
+            optimize_model = False
+            kwargs["modules_to_not_convert"] = ["lm_head"]
```

Comment:
This should be specified by the user?

@yangw1234 - do we always need this? Existing QLoRA examples does not have that.

Reply:
This is the default of bitsandbytes. I think we'd better also set it this way.

Reply:
bitsandbytes will set `kwargs["modules_to_not_convert"] = ["lm_head"]` by default?

Reply:
Yes, I think so. https://github.com/huggingface/transformers/blob/v4.36.2/src/transformers/integrations/bitsandbytes.py#L244-L248

Reply:
AWQ also has similar behavior:
https://github.com/huggingface/transformers/blob/v4.36.2/src/transformers/modeling_utils.py#L3538-L3548